### PR TITLE
Filter by level importancy

### PR DESCRIPTION
**Pull request by Fatimata-dev**
I made a method to filter by level of importance then to avoid code repetition I used a global method in task-list which takes a string parameter then depending on the element in parameter (string, date or enum)

### CODE REVIEW

**Review by sirinebettaieb — APPROVED** [review]
i tested and it's ok

File: src/app/components/task-list/task-list.component.ts [hunk 1]
```diff
@@ -21,9 +21,10 @@ export class TaskListComponent {
   private taskService = inject(TaskService);
   taskList : Task[] = this.taskService.getTasks();
   filteredTasks : Task[] = [];
-
+  levelImportany:ImportancyLevel[]= new Array<ImportancyLevel>();
   ngOnInit():void{
      this.filteredTasks = this.taskList;
+     this.levelImportany = this.taskList.map((task) => task.importancyLevel);
```
Comment (Benja-san) [inline src/app/components/task-list/task-list.component.ts]: importancy

Comment (Benja-san) [reply src/app/components/task-list/task-list.component.ts]: what is the purpose of this map ?

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/components/task-list/task-list.component.ts (hunk 1 above)
Comment (Fatimata-dev) [reply src/app/components/task-list/task-list.component.ts]: that was for a test

**Review by Benja-san — APPROVED** [review]

**Pull request merged**